In [ ]:
# Import the packages that we will be using
import pandas as pd                 # For data handling

#pd.set_option('display.max_columns', 100) # Show all columns when looking at dataframe


In [ ]:
# Define where you are running the code: colab or local
RunInColab          = True     # (False: no  | True: yes)

# If running in colab:
if RunInColab:
    # Mount your google drive in google colab
    from google.colab import drive
    drive.mount('/content/drive')

    # Find location
    #!pwd
    #!ls
    #!ls "/content/drive/My Drive/Colab Notebooks/MachineLearningWithPython/"

    # Define path del proyecto
    Ruta            = "/content/drive/MyDrive/NOTEBOOK/TC1002S/NotebooksStudents/A01643954/"

else:
    # Define path del proyecto
    Ruta            = ""

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
# url string that hosts our .csv file
url = Ruta + "datasets/iris/iris.csv"

# Read the .csv file and store it as a pandas Data Frame
df = pd.read_csv(url)
df

5.1  3.5  1.4  0.2     Iris-setosa
0    4.9  3.0  1.4  0.2     Iris-setosa
1    4.7  3.2  1.3  0.2     Iris-setosa
2    4.6  3.1  1.5  0.2     Iris-setosa
3    5.0  3.6  1.4  0.2     Iris-setosa
4    5.4  3.9  1.7  0.4     Iris-setosa
..   ...  ...  ...  ...             ...
144  6.7  3.0  5.2  2.3  Iris-virginica
145  6.3  2.5  5.0  1.9  Iris-virginica
146  6.5  3.0  5.2  2.0  Iris-virginica
147  6.2  3.4  5.4  2.3  Iris-virginica
148  5.9  3.0  5.1  1.8  Iris-virginica

[149 rows x 5 columns]

In [ ]:
df.describe()

5.1         3.5         1.4         0.2
count  149.000000  149.000000  149.000000  149.000000
mean     5.848322    3.054362    3.773826    1.206040
std      0.828594    0.435810    1.760543    0.760354
min      4.300000    2.000000    1.000000    0.100000
25%      5.100000    2.800000    1.600000    0.300000
50%      5.800000    3.000000    4.400000    1.300000
75%      6.400000    3.300000    5.100000    1.800000
max      7.900000    4.400000    6.900000    2.500000

In [ ]:
# Check for missing data and column names
missing_data = df.isnull().sum()
column_names = df.columns

print("Missing Data:")
print(missing_data)

print("\nColumn Names:")
print(column_names)

Missing Data:
5.1            0
3.5            0
1.4            0
0.2            0
Iris-setosa    0
dtype: int64

Column Names:
Index(['5.1', '3.5', '1.4', '0.2', 'Iris-setosa'], dtype='object')


In [ ]:
# Add column names
column_names = ["Longitud del sépalo", "Ancho del sépalo", "Longitud del pétalo", "Ancho del pétalo", "Especies (Tipo de flor)"]
df.columns = column_names

# Print DataFrame with column names
df

In [ ]:
# Create a new dataset containing only petal width, petal length, and the type of flower
petal_data = df[["Longitud del pétalo", "Ancho del pétalo", "Especies (Tipo de flor)"]]

# Create a new dataset containing only sepal width, sepal length, and the type of flower
sepal_data = df[["Longitud del sépalo", "Ancho del sépalo", "Especies (Tipo de flor)"]]

# Print the datasets
print("Petal Data:")
print(petal_data)

print("\nSepal Data:")
print(sepal_data)

Petal Data:
     Longitud del pétalo  Ancho del pétalo Especies (Tipo de flor)
0                    1.4               0.2             Iris-setosa
1                    1.3               0.2             Iris-setosa
2                    1.5               0.2             Iris-setosa
3                    1.4               0.2             Iris-setosa
4                    1.7               0.4             Iris-setosa
..                   ...               ...                     ...
144                  5.2               2.3          Iris-virginica
145                  5.0               1.9          Iris-virginica
146                  5.2               2.0          Iris-virginica
147                  5.4               2.3          Iris-virginica
148                  5.1               1.8          Iris-virginica

[149 rows x 3 columns]

Sepal Data:
     Longitud del sépalo  Ancho del sépalo Especies (Tipo de flor)
0                    4.9               3.0             Iris-setosa
1            

In [ ]:
# Define a mapping dictionary for encoding flower types
flower_mapping = {'Iris-setosa': 0, 'Iris-versicolor': 1, 'Iris-virginica': 2}

# Create a new column in the original DataFrame with encoded flower types
df['Encoded_Flower_Type'] = df['Especies (Tipo de flor)'].map(flower_mapping)

# Create a new dataset containing sepal width, sepal length, and the encoded flower type
encoded_sepal_data = df[["Longitud del sépalo", "Ancho del sépalo", "Encoded_Flower_Type"]]

# Print the new dataset
print("Encoded Sepal Data:")
print(encoded_sepal_data)

Encoded Sepal Data:
     Longitud del sépalo  Ancho del sépalo  Encoded_Flower_Type
0                    4.9               3.0                    0
1                    4.7               3.2                    0
2                    4.6               3.1                    0
3                    5.0               3.6                    0
4                    5.4               3.9                    0
..                   ...               ...                  ...
144                  6.7               3.0                    2
145                  6.3               2.5                    2
146                  6.5               3.0                    2
147                  6.2               3.4                    2
148                  5.9               3.0                    2

[149 rows x 3 columns]


Nombre de cada columna:
Columna 1: Longitud del sépalo
Columna 2: Ancho del sépalo
Columna 3: Longitud del pétalo
Columna 4: Ancho del pétalo

Tipo de cada columna:
Columna 1 (Longitud del sépalo): Numérica, continua.
Columna 2 (Ancho del sépalo): Numérica, continua.
Columna 3 (Longitud del pétalo): Numérica, continua.
Columna 4 (Ancho del pétalo): Numérica, continua.

Mínimo, máximo, media, promedio, mediana, desviación estándar:
Longitud del sépalo:
Mínimo: 4.3
Máximo: 7.9
Media: 5.848322
Mediana: 5.8
Desviación estándar: 0.828594

Ancho del sépalo:
Mínimo: 2.0
Máximo: 4.4
Media: 3.054362
Mediana: 3.0
Desviación estándar: 0.435810

Longitud del pétalo:
Mínimo: 1.0
Máximo: 6.9
Media: 3.773826
Mediana: 4.4
Desviación estándar: 1.760543

Ancho del pétalo:
Mínimo: 0.1
Máximo: 2.5
Media: 1.206040
Mediana: 1.3
Desviación estándar: 0.760354
